test/e2e/service.go

```diff
@@ -1337,39 +1337,57 @@ var _ = framework.KubeDescribe("Services", func() {
 		framework.CheckReachabilityFromPod(true, normalReachabilityTimeout, namespace, dropPodName, svcIP)
 	})
 
-	It("should be able to create an internal type load balancer on Azure [Slow]", func() {
-		framework.SkipUnlessProviderIs("azure")
+	It("should be able to create an internal type load balancer [Slow]", func() {
```

Comment:
I would prefer to just split this into two different tests if we are going to have so much provider specific logic